### PR TITLE
read ip addr table with json for easier machine parsing

### PR DESCRIPTION
**Pull request by addyess**

### CODE REVIEW

File: lib/charms/layer/kubernetes_common.py [hunk 1]
```diff
@@ -733,14 +733,45 @@ def enable_ipv6_forwarding():
...
+    addrs = []
+    for line in output.decode("utf8").splitlines():
+        intf, state, *intf_addrs = line.split()
+        addr_info = [dict(interface=_as_interface(addr_str)) for addr_str in intf_addrs]
+        addrs.append(dict(ifname=intf, operstate=state, addr_info=addr_info))
+    return addrs
+
+
+def get_bind_addrs(ipv4=True, ipv6=True):
+    bind_addrs = _get_bind_addrs_json() or _get_bind_addrs_table()
```
Comment (stonepreston) [inline lib/charms/layer/kubernetes_common.py]: can we not rely solely on the json-based method for all cases?

Comment (addyess) [reply lib/charms/layer/kubernetes_common.py]: i believe that `series=bionic` doesn't support this `--json` output.. but i could be wrong

Comment (addyess) [reply lib/charms/layer/kubernetes_common.py]: OO... I stand corrected.  This does work on bionic.  It must be `xenial` where this doesn't work.  But we've recently dropped xenial soooo.... i could go for using the json version only

@Cynerva or @kwmonroe thoughts?

Comment (addyess) [reply lib/charms/layer/kubernetes_common.py]: this logic gets much easier when we don't have to worry about reading the table output from `ip addr show`

Comment (Cynerva) [reply lib/charms/layer/kubernetes_common.py]: Sounds good to me. :+1:

Comment (addyess) [reply lib/charms/layer/kubernetes_common.py]: ok, i'll rip out the parsing of the table, and only use the json formatted output

**Review by kwmonroe — APPROVED** [review]

**Pull request merged**